internal/resources/kustomization/resource_kustomization_test.go

```diff
@@ -0,0 +1,273 @@
+	testhelper "github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/testing"
+)
+
+// nolint: gosec
+const (
+	kustomizationResource      = ResourceName
+	kustomizationResourceVar   = "test_kustomizationy"
+	kustomizationDataSourceVar = "test_data_source_kustomizationy"
+	kustomizationNamePrefix    = "tf-kustomization-test"
+	namespaceName              = "tanzu-continuousdelivery-resources"
+
+	gitRepositoryResource    = gitrepositoryhelper.ResourceName
+	gitRepositoryResourceVar = "test_git_repository"
+	gitRepositoryNamePrefix  = "tf-gr-test"
+)
+
+type testAcceptanceConfig struct {
+	Provider                  *schema.Provider
+	KustomizationResource     string
+	KustomizationResourceVar  string
+	KustomizationResourceName string
+	KustomizationName         string
+	ScopeHelperResources      *commonscope.ScopeHelperResources
+	GitRepositoryResource     string
+	GitRepositoryResourceVar  string
+	GitRepositoryName         string
+}
+
+func testGetDefaultAcceptanceConfig(t *testing.T) *testAcceptanceConfig {
+	return &testAcceptanceConfig{
+		Provider:                  initTestProvider(t),
+		KustomizationResource:     kustomizationResource,
+		KustomizationResourceVar:  kustomizationResourceVar,
+		KustomizationResourceName: fmt.Sprintf("%s.%s", kustomizationResource, kustomizationResourceVar),
+		KustomizationName:         acctest.RandomWithPrefix(kustomizationNamePrefix),
+		ScopeHelperResources:      commonscope.NewScopeHelperResources(),
+		GitRepositoryResource:     gitRepositoryResource,
+		GitRepositoryResourceVar:  gitRepositoryResourceVar,
+		GitRepositoryName:         acctest.RandomWithPrefix(gitRepositoryNamePrefix),
+	}
+}
+
+func TestAcceptanceForKustomizationResource(t *testing.T) {
+	testConfig := testGetDefaultAcceptanceConfig(t)
+
+	t.Log("start kustomization resource acceptance tests!")
+
+	// Test case for kustomization resource.
+	resource.Test(t, resource.TestCase{
+		PreCheck:          testhelper.TestPreCheck(t),
+		ProviderFactories: testhelper.GetTestProviderFactories(testConfig.Provider),
+		CheckDestroy:      nil,
+		Steps: []resource.TestStep{
+			{
+				PreConfig: func() {
+					if testConfig.ScopeHelperResources.Cluster.KubeConfigPath == "" {
+						t.Skip("KUBECONFIG env var is not set for cluster scoped kustomization acceptance test")
+					}
+				},
+				Config: testConfig.getTestKustomizationResourceBasicConfigValue(commonscope.ClusterScope),
+				Check:  testConfig.checkKustomizationResourceAttributes(commonscope.ClusterScope),
+			},
+			{
+				Config: testConfig.getTestKustomizationResourceBasicConfigValue(commonscope.ClusterScope, WithPruneEnabled(true), WithInterval("10m")),
+				Check:  testConfig.checkKustomizationResourceAttributes(commonscope.ClusterScope),
+			},
+			{
+				Config: testConfig.getTestKustomizationResourceBasicConfigValue(commonscope.ClusterGroupScope),
+				Check:  testConfig.checkKustomizationResourceAttributes(commonscope.ClusterGroupScope),
+			},
```

Comment:
Add similar update for cluster group scope as well.

Reply:
added  and tested PTAL

`
=== RUN   TestAcceptanceForKustomizationResource
    resource_kustomization_test.go:73: start kustomization resource acceptance tests!
TMC resources applied to the cluster successfully
    resource_kustomization_test.go:106: kustomization resource acceptance test complete
--- PASS: TestAcceptanceForKustomizationResource (224.57s)
PASS
ok      github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/kustomization     226.067s
`